### DISCUSSION TITLE
tailscale optionen für exit-node und subnet

### DISCUSSION THREAD

**Discussion by konne** [OpenCCU/OpenCCU#2402] [Development]
Hi,

aktuell announced tailscale auf der CCU sich selbst als exit-node und das lokale subnet.
Ich finde im standard sollte es nichts announcen.
@jens-maus wie fändest du es wenn man im webinterface noch zwei weitere optionskästchen hätte die man dann anklicken kann.
Also eins für exit node und eins für das subnet und nur dann werden diese parameter in den ARGS für tailscale up mitgegeben.

Wenn du mir noch einen Tip gibst, wo / wie ich am besten diesen Status speichere und dann nutze, würde ich mich mal an einem PR versuchen.
Du kannst mir auch einfach einen Hinweis geben, bei welchem anderen Packet du das machst, dann kann ich mich dort im Quelltext einlesen.

BG

**Comment by Baxxy13** [comment]
Du kannst die "TAILSCALE_UP_ARGS" überschreiben indem du deine Eigenen in der `/usr/local/etc/config/tailscaleEnabled` ablegst.
z.B. `TAILSCALE_UP_ARGS="--hostname=baxxys-Tail-RM"`

Das Problem ist das diese Datei erst bei Aktivierung von Tailscale angelegt wird und somit das erste `tailscale up` mit den default-Parametern erfolgt.
Um da wieder rauszukommen brauchts erstmal ein `tailscale up --reset`, dann machen wir ein `/etc/init.d/S46tailscaled stop`, tragen unsere eigenen Parameter in die `tailscaleEnabled` ein und anschließend ein `/etc/init.d/S46tailscaled start`.
Voila, Tailscale läuft mit eigener Config.

Umständlich und wenig intuitiv.

Meide Idee wäre die eigenen "TAILSCALE_UP_ARGS" in eine separate Datei auszulagern welche idealerweise über ein Eingabefeld in der WebUI befüllt werden kann. Also keine Checkboxen sondern ein Textfeld indem ich meine Parameter reinschreibe.

Als rudimentärer Coder habe ich dazu erstmal den "Start-Bereich" in der `S46tailscaled` um einen Block erweitert.

```
                # check if the user has set up own startup-args
                if [[  -e /usr/local/tailscale_up_args ]]; then
                    TAILSCALE_UP_ARGS=$( /bin/cat /usr/local/tailscale_up_args )
                fi
```
Das funktionierte beim Test gestern ganz gut.

Vielleicht mal so als möglicher Ansatz.

Edit: um Tailscale ohne extra Parameter zu starten trägt man in die `tailscaleEnabled` folgendes ein: `TAILSCALE_UP_ARGS=""`
